norddrop/src/ffi/mod.rs

```diff
@@ -260,6 +260,78 @@ pub extern "C" fn norddrop_stop(dev: &norddrop) -> norddrop_result {
     result.unwrap_or(norddrop_result::NORDDROP_RES_ERROR)
 }
 
+/// Purge transfers with the given id(s) from the database, accepts a JSON array
+/// of strings
+#[no_mangle]
+pub extern "C" fn norddrop_purge_transfers(
+    dev: &norddrop,
+    txids: *const c_char,
+) -> norddrop_result {
+    let result = panic::catch_unwind(move || {
+        let mut dev = match dev.0.lock() {
+            Ok(inst) => inst,
+            Err(poisoned) => poisoned.into_inner(),
+        };
+
+        let txids = {
+            if txids.is_null() {
+                return norddrop_result::NORDDROP_RES_INVALID_STRING;
+            }
+
+            ffi_try!(unsafe { CStr::from_ptr(txids) }.to_str())
+        };
+
+        dev.purge_transfers(txids)
+            .norddrop_log_result(&dev.logger, "norddrop_purge_transfers")
+    });
+
+    result.unwrap_or(norddrop_result::NORDDROP_RES_ERROR)
+}
+
+/// Purge all transfers that are older than the given timestamp from the
+/// database. Accepts a UNIX timestamp in seconds
+#[no_mangle]
+pub extern "C" fn norddrop_purge_transfers_until(
+    dev: &norddrop,
+    until_timestamp: i64,
+) -> norddrop_result {
+    let result = panic::catch_unwind(move || {
+        let mut dev = match dev.0.lock() {
+            Ok(inst) => inst,
+            Err(poisoned) => poisoned.into_inner(),
+        };
+
+        dev.purge_transfers_until(until_timestamp)
+            .norddrop_log_result(&dev.logger, "norddrop_purge_transfers_until")
+    });
+
+    result.unwrap_or(norddrop_result::NORDDROP_RES_ERROR)
+}
+
+/// Get all transfers since the given timestamp from the database. Accepts a
+/// UNIX timestamp in seconds
+#[no_mangle]
+pub extern "C" fn norddrop_get_transfers_since(
+    dev: &norddrop,
+    since_timestamp: i64,
+) -> *mut c_char {
```

Comment:
I'm unsure if returning `*mut c_char` is good here. With this in the `new_transfer` function, we already shoot ourselves in the foot. Because getting `null` can mean anything. 
Maybe we can do something like `transfers_since(dev: &norddrop, since: i64, out_transfers: **mut c_char) -> norddrop_result` and write the `char *` to the `out_transfers` on success? I'm not sure how swig will handle that TBH

Reply:
I think this is a double-edged sword. The API would be better but I smell binding issues and problems by accepting pointer to pointer. This is fine to be left as tech debt for future revisions when we already have features going

Reply:
Yeah, it's probably true. But since it's potentially breaking change in the future I would at least try to see if Swig is a problem here.
If yes, then it's better to abandon the idea

Reply:
I'll have a look if it's a trivial change or not :+1:

Reply:
I think it's hard to quickly say if it's a good idea as Golang bindings have shown after some time that it causes trouble. Or C#. I suggest not blocking this PR due to this however exploring this idea for some small amount of time after it's been landed :)

Reply:
I'll create a ticket for this as a possible improvement then, to keep it in the back of our minds